ingest/indexer/service/client/pubsub_client.go

```diff
@@ -33,8 +35,55 @@ func NewPubSubCLient(projectId string, blockTopicId string, transactionTopicId s
 	}
 }
 
+// setIngestedAtField sets the IngestedAt field of a message to the given time.
+func (p *PubSubClient) setIngestedAt(message any, ingestedAt time.Time) error {
```

Comment:
I'm not a huge fan of reflection due to added complexity. Is it possible to explicitly set the "ingestedAt' field for every message?

Alternatively, can we start adding unit tests for every possible message type?

Reply:
Not a big fans of it either but just thought for system attribute like that, it might be convenient for not having to worry about such attribute as long as such attribute is defined in the struct. 

No big deal. I've switched it to explicitly setting it.

Test coverage will be added as we discussed it on 1-1

Thanks!!